File: introductionToAlgorithmCLRS/chp2/InChaptAlgo/insertionSort.java
import java.util.Scanner;
import java.util.Arrays;

public class insertionSort {


    public static int[] sorting (int[] array) {
        int sortedArray[] = array;
        for(int j = 1; j < sortedArray.length; j++) {
            int key = sortedArray[j];
            int i = j - 1 ;
            while (i >-1 && sortedArray[i] > key ) {
                sortedArray[i + 1] = sortedArray[i];
                i = i - 1;
            }
            sortedArray[i+1] = key;
        }
        
        return(sortedArray);
    }
    public static void main(String[] args){

        int size;
        System.out.print("Enter the size of array you wish to have :- ");
        Scanner element = new Scanner(System.in);
        size = element.nextInt();
        int array[] = new int[size];

        System.out.print("Enter the elements of the array : ");

        for (int i = 0; i < size; i++) {
            array[i] = element.nextInt();
        }

        element.close();

        int sortedArray[] = sorting(array);

        System.out.println("New Sorted array is : " + Arrays.toString(sortedArray));

    }
}

File: introductionToAlgorithmCLRS/chp2/InChaptAlgo/insertionSort.c
#include <stdio.h>

int  *insertionSort(int array[], int size) {

    for (int i = 1; i < size; i++) {
        int key = array[i];
        int j = i - 1;
        while (j > -1 && array[j] > key) {
            array[j + 1] = array[j];
            j = j - 1;
        }
        array[j + 1] = key;
    }
    return array;
}
int main() {
    int size;
    printf("Enter the size of array you need : ");
    scanf("%d", &size);
    int array[size];
    printf("Enter the elements of the array \n");
    for (int i = 0 ; i< size; i++) {
        scanf("%d", &array[i]);
    }
    int *sortedArray;
    sortedArray = insertionSort(array, size);

    printf("Sorted array is : [ ");
    for (int i = 0; i < size; i++) {
        printf("%d ", sortedArray[i]);
    }
    printf("]");
}